plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestJoinQueryOnPartitionedColumn.java

```diff
@@ -0,0 +1,94 @@
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public class TestJoinQueryOnPartitionedColumn
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("iceberg")
+                .setSystemProperty("optimize_metadata_queries", "true")
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+
+        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile();
+
+        HdfsConfig hdfsConfig = new HdfsConfig();
+        HdfsConfiguration hdfsConfiguration = new HiveHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of());
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
+
+        HiveMetastore metastore = new FileHiveMetastore(
+                new NodeVersion("test_version"),
+                hdfsEnvironment,
+                new MetastoreConfig(),
+                new FileHiveMetastoreConfig()
+                        .setCatalogDirectory(baseDir.toURI().toString())
+                        .setMetastoreUser("test"));
+
+        queryRunner.installPlugin(new TestingIcebergPlugin(metastore, false));
+        queryRunner.createCatalog("iceberg", "iceberg");
+
+        return queryRunner;
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        assertUpdate("CREATE SCHEMA test_schema");
+        assertUpdate("CREATE TABLE test_schema.test_table (a BIGINT, b TIMESTAMP(6) with time zone, c row(d BIGINT, e BIGINT)) WITH (partitioning = ARRAY['a', 'day(b)'])");
+        assertUpdate("create table test_schema.test_table_2 (foo BIGINT, bar BIGINT)");
+
+        assertUpdate("INSERT INTO test_schema.test_table VALUES (0, CAST('2019-09-08' AS TIMESTAMP(6) with time zone), (1, 1)), " +
+                "(1, CAST('2020-09-09' AS TIMESTAMP(6) with time zone), (1, 1)), " +
+                "(2, CAST('2021-09-09' AS TIMESTAMP(6) with time zone), (1, 1))", 3);
+        assertUpdate("INSERT INTO test_schema.test_table VALUES (15, CAST('2019-09-08' AS TIMESTAMP(6) with time zone), (1, 1)), " +
+                "(20, CAST('2020-09-09' AS TIMESTAMP(6) with time zone), (1, 1)), " +
+                "(50, CAST('2021-09-09' AS TIMESTAMP(6) with time zone), (1, 1))", 3);
+        assertUpdate("INSERT INTO test_schema.test_table_2 VALUES (0,100), (1,101), (15,102), (16,103)", 4);
+    }
+
+    @Test
+    public void testPartitionTable()
```

Comment:
did you consider adding this as a new test method within `AbstractTestIcebergConnectorTest`?
(would require using more unique table names)

or, is query runner setup unique in this test?

Reply:
oh, i have read some test method in the `AbstractTestIcebergConnectorTest` and think this pr may be related to the method `testDayTransformTimestamp`. Added another simple table `t1(foo BIGINT, bar VARCHAR)`, let table `test_day_transform_timestamp` join `t1` here, the test will failed

Reply:
That test doesn't catch the issue because we are testing only the `$partition` table in that test but don't have any tests for partition pruning using stats (`TableStatisticsMaker#dataFileMatches` method).

Reply:
@hashhar Thanks for explain, i want to add some tests for this pr in the `AbstractTestIcebergConnectorTest` and want to test against on `timestamp(6) with timezone`, but seems H2 do not support, can you advise me how to test it? 
cc @findepi  

the slack: https://trinodb.slack.com/archives/CP1MUNEUX/p1622793477213900

Reply:
You can use `.matches` to assert values directly in such cases. See https://github.com/trinodb/trino/blob/9fa8d78fbe63415e9d5898f4d8dd22f91d33e0bc/testing/trino-tests/src/test/java/io/trino/tests/AbstractQueryAssertionsTest.java#L195 for examples.